ue4docker/dockerfiles/ue4-build-prerequisites/windows/install-prerequisites.bat

```diff
@@ -10,24 +10,27 @@ call refreshenv
 @rem Forcibly disable the git credential manager
 git config --system credential.helper "" || goto :error
 
-@rem Install the Visual Studio 2017 Build Tools workloads and components we need
-@rem (Note that we use the Visual Studio 2019 installer here because the old installer now breaks, but explicitly install the VS2017 Build Tools)
+set VISUAL_STUDIO_BUILD_NUMBER=%~1
+
+@rem Install the Visual Studio Build Tools workloads and components we need
+@rem (Note that we use the Visual Studio 2019 installer even for Visual Studio 2017 here because the old installer now breaks)
 curl --progress-bar -L "https://aka.ms/vs/16/release/vs_buildtools.exe" --output %TEMP%\vs_buildtools.exe || goto :error
 %TEMP%\vs_buildtools.exe --quiet --wait --norestart --nocache ^
 	--installPath C:\BuildTools ^
-	--channelUri "https://aka.ms/vs/15/release/channel" ^
-	--installChannelUri "https://aka.ms/vs/15/release/channel" ^
-	--channelId VisualStudio.15.Release ^
+	--channelUri "https://aka.ms/vs/%VISUAL_STUDIO_BUILD_NUMBER%/release/channel" ^
+	--installChannelUri "https://aka.ms/vs/%VISUAL_STUDIO_BUILD_NUMBER%/release/channel" ^
+	--channelId VisualStudio.%VISUAL_STUDIO_BUILD_NUMBER%.Release ^
 	--productId Microsoft.VisualStudio.Product.BuildTools ^
 	--locale en-US ^
 	--add Microsoft.VisualStudio.Workload.VCTools ^
 	--add Microsoft.VisualStudio.Workload.MSBuildTools ^
 	--add Microsoft.VisualStudio.Component.NuGet ^
 	--add Microsoft.VisualStudio.Component.Roslyn.Compiler ^
+	--add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 ^
 	--add Microsoft.VisualStudio.Component.Windows10SDK.17763 ^
-	--add Microsoft.Net.Component.4.5.TargetingPack ^
-	--add Microsoft.Net.Component.4.6.2.SDK ^
-	--add Microsoft.Net.Component.4.6.2.TargetingPack
+	--add Microsoft.Net.ComponentGroup.4.6.2.DeveloperTools ^
+	--add Microsoft.Net.Component.4.5.TargetingPack
```

Comment:
I'm curious why we need the 4.5 Targeting Pack. (I realise it's not new, just changed lines, hence only curious). Particularly because I don't understand how a TargetingPack works without the relevant SDK installed as well.

Reply:
I'm not a .NET expert, but from what I've read, Targeting Pack is something that contains .NET DLLs + some MSBuild config files so MSBuild understands how to build .NET applications against specified .NET.

And SDK is a different thing. Most noticeable artifacts it installs is C++ headers under `C:\Program Files (x86)\Windows Kits\NETFXSDK.

And now the weird bit: 
1. VS2017 has 4.6.2 SDK
2. VS2019 has 4.6.2 SDK
3. But VS2019 *Build Tools* doesn't have 4.6.2 SDK. When you ask it to install `Microsoft.Net.ComponentGroup.4.6.2.DeveloperTools`, it will actually install 4.6.2 Targeting Pack + *4.8* SDK. I'm not sure whether will will work at all, running a build currently.

Reply:
Interesting. The VS2019 Build Tools install is doing is what I would have _expected_: that the Targeting Pack for 4.N + _latest_ 4.x SDK would let you build targeting the .NET Framework 4.N, so you didn't have to keep multiple copies of the SDK around when each was a superset of the previous anyway.

That was my original vague understanding of .NET Framework development and the Targeting Packs, but I'd never actually had to put that to the test.

So perhaps Build Tools is right, but the UI installers let you be inefficient if you want by multi-installing SDKs or installing only old versions of the SDKs, and they've now trimmed that from the Build Tools installer as not necessary.

Reply:
> I'm curious why we need the 4.5 Targeting Pack

I don't remember actually. I discovered that it is required when was working on #144. Some of UE4 C# tools builds against 4.5, but I forgot which tool exactly.